public/app/core/reducers/location.ts

```diff
@@ -29,6 +30,24 @@ export const locationReducer = (state: LocationState = initialState, action: Act
 
     if (payload.partial) {
       query = _.defaults(query, state.query);
+    }
```

Comment:
@hugohaggmark -- I'm sure this breaks all kinds of redux rules... is there a better place for this to live?